Fix error messages for failed assertions in Scripts

In SDK 1.16.0 the script

```
t : Script ()
t = assert False
```

leads to the Script results

```
"Scenario service backend error: BErrorClient (ClientIOError (GRPCIOBadStatusCode StatusUnknown (StatusDetails {unStatusDetails = \"\"})))"
```

I believe this is a regression related to Exception Handling. The reason I think so is that the Language Server Output contains

```
2021-08-12 11:14:35.50 [ERROR]  [LanguageServer ScenarioService]  [] 
SCENARIO SERVICE STDERR: Error: Unhandled exception: DA.Exception.AssertionFailed:AssertionFailed@3f4deaf1{ message = "Assertion failed" }
```

--- a/compiler/damlc/tests/src/DA/Test/ScriptService.hs
+++ b/compiler/damlc/tests/src/DA/Test/ScriptService.hs
@@ -901,6 +901,9 @@ main =
                     , "           error \"\""
                     , "         catch"
                     , "           (GeneralError _) -> pure ()"
+                    , "    choice Fail : ()"
+                    , "      controller p"
+                    , "      do assert False"
                     , "test = do"
                     , "  p <- allocateParty \"p\""
                     , "  cid <- submit p $ createCmd (T p)"
@@ -908,9 +911,15 @@ main =
                     , "  r <- query @T p"
                     , "  r === [(cid, T p)]"
                     , "  pure ()"
+                    , "unhandledOffLedger = script $ assert False"
+                    , "unhandledOnLedger = script $ do"
+                    , "  p <- allocateParty \"p\""
+                    , "  submit p $ createAndExerciseCmd (Helper p) Fail"
                     ]
                 expectScriptSuccess rs (vr "test") $ \r ->
                   matchRegex r "Active contracts:  #0:0\n"
+                expectScriptFailure rs (vr "unhandledOffLedger") $ \r -> matchRegex r "Unhandled exception"
+                expectScriptFailure rs (vr "unhandledOnLedger") $ \r -> matchRegex r "Unhandled exception"
             ]
   where
     scenarioConfig = SS.defaultScenarioServiceConfig {SS.cnfJvmOptions = ["-Xmx200M"]}

--- a/compiler/scenario-service/server/src/main/scala/com/digitalasset/daml/lf/scenario/Context.scala
+++ b/compiler/scenario-service/server/src/main/scala/com/digitalasset/daml/lf/scenario/Context.scala
@@ -223,6 +223,9 @@ class Context(val contextId: Context.ContextId, languageVersion: LanguageVersion
           )
         )
       case Failure(e: Error) => handleFailure(e)
+      case Failure(e: Runner.InterpretationError) => {
+        handleFailure(Error.RunnerException(e.error))
+      }
       case Failure(e: ScriptF.FailedCmd) =>
         e.cause match {
           case e: Error => handleFailure(e)

